HPCC-14636 Add parallel quick sorts to roxie (for comparison)

ghalliday opened the pull request:
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

ghalliday committed:
HPCC-14636 Add parallel quick sorts to roxie (for comparison)

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
@richardkchapman please review

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented on common/thorhelper/roxiehelper.hpp:
```diff
@@ -104,6 +104,7 @@ typedef enum { heapSortAlgorithm, insertionSortAlgorithm,
               mergeSortAlgorithm, spillingMergeSortAlgorithm,
               parallelMergeSortAlgorithm, spillingParallelMergeSortAlgorithm,
               tbbQuickSortAlgorithm, tbbStableQuickSortAlgorithm,
```
Are tbbQuickSort and tbbStableQuickSort not parallel?

Perhaps some comments on this enum describing what is what would be helpful?

What algo is used if they just indicate they want a parallel sort without being specific about which one (can they do that?)

richardkchapman commented:
@ghalliday Couple of questions

ghalliday commented:
@richardkchapman updated to reflect comments.  I also changed the spill logic to allow it to be set to false.
I added a couple of jiras to cover cleaning up parallel/stable/spilling options.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
